### ISSUE TITLE
`fossa` does not resolve bower directory from `.bowerrc` config

### ISSUE THREAD

**Issue by wolf31o2** [fossas/fossa-cli#125]
I am attempting to switch to provided builds for a small javascript service. The build process is done via custom scripts, but it's not very complex or custom.

build.sh:
```
#!/usr/bin/env bash
# This is run during the pipeline build - it builds the code that gets run eventually
set -e

npm install --unsafe-perm
npm run build
```

After this, unit tests are run.
unit.sh:
```
#!/usr/bin/env bash
# This is run during the pipeline build testing step
set -e

npm run lint
npm run test

### HACK: Using this repository to play with FOSSA. Sorry for the noise.
export PATH=${PATH}:${WORKSPACE}/node_modules/bower/bin
fossa -o --allow-unresolved --debug
```

Here's the configuration.
.fossa.yml:
```yaml
version: 1
cli:
  server: https://app.fossa.io
  project: github.com/ApplauseAQI/auth-service
  fetcher: custom
analyze:
  modules:
  - name: auth-service
    path: bower.json
    type: bower
  - name: auth-service
    path: package.json
    type: nodejs
```

And finally, the relevant log section.

Logs:
```
12:16:22 2018-04-13T16:16:22.739Z DEBUG config:config/init.go:122/New Configuration initialized: config.CLIConfig{APIKey:"", Fetcher:"custom", Project:"github.com/ApplauseAQI/auth-service", Revision:"d29ee34067da1fd531e1edb01d8499cc5538bfc1", Endpoint:"https://app.fossa.io", Modules:[]module.Config{module.Config{Name:"auth-service", Path:"bower.json", Type:"bower"}, module.Config{Name:"auth-service", Path:"package.json", Type:"nodejs"}}, Debug:true, DefaultCmd:config.DefaultConfig{Build:false}, AnalyzeCmd:config.AnalyzeConfig{Output:true, AllowUnresolved:true}, BuildCmd:config.BuildConfig{Force:false}, TestCmd:config.TestConfig{Timeout:0}, UploadCmd:config.UploadConfig{Locators:false, Data:""}, ReportCmd:config.ReportConfig{Type:""}, ConfigFilePath:".fossa.yml"}
12:16:22 2018-04-13T16:16:22.934Z DEBUG update:main/update.go:34/checkUpdate checking version for updates (0.5.0 -> &{0.5.0 https://github.com/fossas/fossa-cli/releases/download/v0.5.0/fossa-cli_0.5.0_linux_amd64.tar.gz %!s(int=4281298) %!s(int64=6668517) https://github.com/fossas/fossa-cli/releases/tag/v0.5.0 ## Changelog
12:16:22 
12:16:22 2954eee Merge pull request #121 from fossas/feat/nuget-support
12:16:22 8d58e9c test(builders): add nuget fixtures
12:16:22 6884192 doc(readme): update readme
12:16:22 99d3f8c Merge branch 'master' into feat/nuget-support
12:16:22 1dbda7d feat(build): turn built module error into a warning
12:16:22 bc5811c doc(builders): add nuget docs
12:16:22 377a05a feat(builders): add nuget lockfile parsing
12:16:22 843299a feat(builders): add nuget support
12:16:22 c168cce chore(deps): Update dependencies
12:16:22 5e146c5 feat(builders): Add Pip support
12:16:22 
12:16:22 ---
12:16:22 Automated with [GoReleaser](https://github.com/goreleaser)
12:16:22 Built with go version go1.10 darwin/amd64
12:16:22  v0.5.0 2018-03-29 03:23:35 +0000 UTC fossas fossa-cli})
12:16:22 2018-04-13T16:16:22.934Z WARNING init:main/init.go:68/doInit 2 module(s) found in config file (`.fossa.yml`); skipping initialization.
12:16:22 ⣾ Running build analysis (1/2): auth-service                                              2018-04-13T16:16:22.935Z DEBUG main:main/main.go:156/resolveModuleConfig Resolving moduleConfig: module.Config{Name:"auth-service", Path:"bower.json", Type:"bower"}
12:16:22 2018-04-13T16:16:22.935Z DEBUG main:main/main.go:167/resolveModuleConfig Got bower module.
12:16:22 2018-04-13T16:16:22.935Z DEBUG main:main/main.go:181/resolveModuleConfig Resolved moduleConfig to: &builders.BowerBuilder{NodeCmd:"", NodeVersion:"", BowerCmd:"", BowerVersion:""}, module.Module{Name:"auth-service", Type:"bower", Target:"/home/centos/jenkins/workspace/auth-service-pr/bower.json", Dir:"/home/centos/jenkins/workspace/auth-service-pr", Project:"", Revision:"", Value:interface {}(nil)}
12:16:22 2018-04-13T16:16:22.935Z DEBUG bower:builders/bower.go:50/Initialize Initializing Bower builder...
12:16:22 2018-04-13T16:16:22.935Z DEBUG common:builders/common.go:154/whichWithResolver Tried resolving `` but did not work: &os.PathError{Op:"fork/exec", Path:"", Err:0x2} ""
12:16:22 2018-04-13T16:16:22.94Z DEBUG common:builders/common.go:154/whichWithResolver Tried resolving `` but did not work: &os.PathError{Op:"fork/exec", Path:"", Err:0x2} ""
12:16:23 ⣽ Running build analysis (1/2): auth-service                                              ⣻ Running build analysis (1/2): auth-service                                              ⢿ Running build analysis (1/2): auth-service 2018-04-13T16:16:23.294Z DEBUG bower:builders/bower.go:68/Initialize Done initializing Bower builder: &builders.BowerBuilder{NodeCmd:"node", NodeVersion:"v6.12.3\n", BowerCmd:"bower", BowerVersion:"1.8.2\n"}
12:16:23 2018-04-13T16:16:23.294Z DEBUG bower:builders/bower.go:126/IsBuilt Checking Bower build: module.Module{Name:"auth-service", Type:"bower", Target:"/home/centos/jenkins/workspace/auth-service-pr/bower.json", Dir:"/home/centos/jenkins/workspace/auth-service-pr", Project:"", Revision:"", Value:interface {}(nil)} true
12:16:23 2018-04-13T16:16:23.294Z DEBUG bower:builders/bower.go:135/IsBuilt Done checking Bower build: false
12:16:23                                              2018-04-13T16:16:23.294Z CRITICAL main:main/main.go:243/defaultCmd Module auth-service does not appear to be built. Try first running your build or `fossa build`, and then running `fossa`.
```

**Comment by ilikebits** [comment]
Thanks for this! I'll take a closer look at this tonight -- would appreciate it greatly if you had a minimal error case you could share.

Here's the high-level overview of the problem: when we analyse your code, we first check whether or not a build is complete. If a completed build is not detected, we throw an error. (Our build completion detection seems to have more false negatives than expected -- we might change this to a warning, and only throw an error if the user sets an opt-in flag.) For Bower, the code is [here](https://github.com/fossas/fossa-cli/blob/master/builders/bower.go#L125) -- it basically checks for the existence of a `bower_components` folder in the target directory.

Action items fixing this issue:

1. Make incomplete builds a warning instead of an error, or provide a method for suppressing the error.
2. Improve our Bower build completion detection -- in particular, building to component folders that are not `bower_components`.

**Comment by wolf31o2** [comment]
So... the problem is we have this:

.bowerrc:
```json
{
  "directory": "public/bower_components"
}
```

**Comment by xizhao** [comment]
Thanks for the background Chris -- I updated the issue title to reflect the new status.  

This seems to be because we hardcode the `bower_components` module directory here:

https://github.com/fossas/fossa-cli/blob/master/builders/bower.go#L130 

It seems like the best solution here is either to glob for a `bower_components` subdirectory like we do in the analyze step, or to check for a `.bowerrc` file, parse it, and resolve the directory.

I assume the analyze step should be working properly, given we glob for the folder prefixed with a `**` before we start traversing for modules.

https://github.com/fossas/fossa-cli/blob/master/builders/bower.go#L97

**Comment by xizhao** [comment]
@wolf31o2 I've submitted a PR that fixes this, tested in my local environment 

@ilikebits please review, merge and release.

**Comment by xizhao** [comment]
Released in [v0.5.1](https://github.com/fossas/fossa-cli/releases/tag/v0.5.1), run `fossa update` to grab it.
